service/history/api/recordworkflowtaskstarted/api.go

```diff
@@ -74,6 +73,10 @@ func Invoke(
 				//  - Speculative WFT is lost (ScheduleToStart timeout for speculative WFT will recreate it).
 				return nil, serviceerror.NewNotFound("Workflow task not found.")
 			}
+			if workflowTask.Stamp != mutableState.GetExecutionInfo().GetWorkflowTaskStamp() {
```

Comment:
I'm a bit confused. `workflowTask.Stamp` seems to be [constructed from current mutable state](https://github.com/stephanos/temporal/blob/213014679a4c8b69d1adf1f6aa9cf5881d583925/service/history/workflow/workflow_task_state_machine.go#L1096). Which means this will never evaluate to true?